### PR TITLE
[Ogoyi]-Sage-A detailed guide on how to create a decentralized marketplace for magazines on the Celo blockchain

### PR DESCRIPTION
**Pull request by Ogoyi**
### Thanks for submitting this tutorial for Celo Sage

Before you hit "Create Pull Request", make sure you have read the [Tutorial Contribution Guidelines](https://docs.celo.org/community/celo-sage/contribution-guidelines) and reviewed the [Sample Tutorial Structure](https://docs.celo.org/community/celo-sage/tutorial-structure).

Please confirm the following before submitting your tutorial for review -

- [x] My tutorial follows the [Tutorial Contribution Guidelines](https://docs.celo.org/community/celo-sage/contribution-guidelines).
- [x] The title of the Pull Request matches the title of my tutorial.
- [x] I've followed the tutorial template structure.
- [x] I've proofread the tutorial for spelling and grammar mistakes.
- [x] I've thoroughly explained the code, with inline code comments and text explanations.
- [x] I've run a linter & prettifier to format my code blocks.
- [x] I've run the tutorial to test it and the code works.

> NOTE: We cannot review your Pull Request if it does not meet the standard outlined above.
> Reviewers - @joenyzio & @viral-sangani

### CODE REVIEW

**Review by Julius170 — COMMENTED** [review]
@Ogoyi Your tutorial looks great and has already been deployed successful, just a few corrections and you'll need to make and it'll be ready for merge

File: blog/03-3-23-A-smart-contract-for-selling-magazine-on-celo-blockchain/magazine-tutorial-sage.md [hunk 1]
```diff
@@ -0,0 +1,480 @@
...
+            magazines[_index].owner,
+            magazines[_index].price
+          ),
+          "Transfer failed."
+        );
+
+        }
+```
+
+The function takes one input parameter _index of type uint, which represents the index of the magazine to be purchased from the magazines mapping.
```
Comment (Julius170) [inline blog/03-3-23-A-smart-contract-for-selling-magazine-on-celo-blockchain/magazine-tutorial-sage.md:324]: encode _index, uint, and index

File: blog/03-3-23-A-smart-contract-for-selling-magazine-on-celo-blockchain/magazine-tutorial-sage.md [hunk 2]
```diff
@@ -0,0 +1,480 @@
...
+                   
+}
+
+  function getmagazineslength() public view returns (uint) {
+        return (magazinesLength);
+  }
+
+```
+
+The function takes one input parameter _index of type uint, which represents the index of the magazine to be unpublished from the magazines mapping.
```
Comment (Julius170) [inline blog/03-3-23-A-smart-contract-for-selling-magazine-on-celo-blockchain/magazine-tutorial-sage.md:354]: encode _index, uint, and index

File: blog/03-3-23-A-smart-contract-for-selling-magazine-on-celo-blockchain/magazine-tutorial-sage.md [hunk 3]
```diff
@@ -0,0 +1,480 @@
...
+
+```
+
+## Contract deployment
+
+To deploy the Magazine smart contract on the Celo blockchain, follow the steps below:
+
+- Install Celo Extension Wallet: Download and install the Celo Extension Wallet from the Google Chrome store. Create a wallet and securely store your key phrase.
+
+- Fund your wallet: Copy your wallet address and paste it into the Celo Faucet. Confirm the transaction to receive Celo tokens in your wallet.
```
Comment (Julius170) [inline blog/03-3-23-A-smart-contract-for-selling-magazine-on-celo-blockchain/magazine-tutorial-sage.md:459]: bold the bullet point "**Fund your wallet**"

File: blog/03-3-23-A-smart-contract-for-selling-magazine-on-celo-blockchain/magazine-tutorial-sage.md [hunk 4]
```diff
@@ -0,0 +1,480 @@
...
+    }
+}
+
+```
+
+## Contract deployment
+
+To deploy the Magazine smart contract on the Celo blockchain, follow the steps below:
+
+- Install Celo Extension Wallet: Download and install the Celo Extension Wallet from the Google Chrome store. Create a wallet and securely store your key phrase.
```
Comment (Julius170) [inline blog/03-3-23-A-smart-contract-for-selling-magazine-on-celo-blockchain/magazine-tutorial-sage.md:457]: bold bullet point

File: blog/03-3-23-A-smart-contract-for-selling-magazine-on-celo-blockchain/magazine-tutorial-sage.md [hunk 5]
```diff
@@ -0,0 +1,480 @@
...
+
+- Fund your wallet: Copy your wallet address and paste it into the Celo Faucet. Confirm the transaction to receive Celo tokens in your wallet.
+
+- Open Remix and create a new Solidity file: Paste the Magazine contract code into the file. Ensure that the Solidity compiler is set to version 0.8.7 or later.
+
+- Compile the contract: Click the "Compile Magazine.sol" button in the Solidity Compiler tab in Remix.
```
Comment (Julius170) [inline blog/03-3-23-A-smart-contract-for-selling-magazine-on-celo-blockchain/magazine-tutorial-sage.md:463]: bold bullet point

File: blog/03-3-23-A-smart-contract-for-selling-magazine-on-celo-blockchain/magazine-tutorial-sage.md [hunk 6]
```diff
@@ -0,0 +1,480 @@
...
+
+- Deploy the contract: In the "Deploy & Run Transactions" tab, select the Celo network from the dropdown menu. Connect your wallet to Remix by clicking "Connect to wallet". Select "Magazine" from the "Contract" dropdown menu. Click the "Deploy" button, confirm the transaction in your wallet, and wait for the transaction to be confirmed on the Celo blockchain.
```
Comment (Julius170) [inline blog/03-3-23-A-smart-contract-for-selling-magazine-on-celo-blockchain/magazine-tutorial-sage.md:465]: bold bullet point

File: blog/03-3-23-A-smart-contract-for-selling-magazine-on-celo-blockchain/magazine-tutorial-sage.md [hunk 7]
```diff
@@ -0,0 +1,480 @@
...
+
+- Interact with the contract: Once the transaction is confirmed, the FloralNft contract will be deployed on the Celo blockchain. You can interact with it using Remix.
```
Comment (Julius170) [inline blog/03-3-23-A-smart-contract-for-selling-magazine-on-celo-blockchain/magazine-tutorial-sage.md:467]: bold bullet point

**Review by Julius170 — COMMENTED** [review]
A really great piece you have here @Ogoyi 

I've made suggestions, you can make corrections and it'll  be ready for merging
And also take note of the syntax I mentioned throughtout the article just incase of your next article

**Review by Ishanpathak1 — APPROVED** [review]
Approved

**Pull request merged**